### DISCUSSION TITLE
meta  tegra-boot-tools

### DISCUSSION THREAD

**Discussion by enhany** [OE4T/meta-tegra#1267] [Q&A] [answered]
Hello,
is it possible to enable full a/b redundancy by default (at compile time), not by running "tegra-boot-control -e" ?

**Comment by madisongh** [comment, marked as answer]
I'm not sure what you mean by "at compile time," but you should be able to set it up from initial flashing of your target device. The redundancy mode is controlled by the slot metadata (SMD), which is stored on the device.  There's a file called `smd_info.cfg` in the L4T kit that is used to generate the SMD contents that gets flashed to the target device.  If you change your build to replace the default file with one that enables A/B redundancy (see the comments in that file), that should take care of it.

**Reply by enhany** [reply]
Is "smd_info.cfg" working with custom tegra-boot-control or with nvidia tegra-redundant-boot ?

**Reply by madisongh** [reply]
Either.

**Reply by enhany** [reply]
Thank you. It works.

Is it possible to use 3 partitions? (a/b/c)? or is it only a/b limited?

**Reply by madisongh** [reply]
Per [NVIDIA's documentation](https://docs.nvidia.com/jetson/archives/l4t-archived/l4t-3273/Tegra%20Linux%20Driver%20Package%20Development%20Guide/bootloader_update.html#), only two slots are supported.
